py/torch_tensorrt/dynamo/runtime/_MutableTorchTensorRTModule.py

```diff
@@ -0,0 +1,274 @@
+        optimization_level: Optional[int] = _defaults.OPTIMIZATION_LEVEL,
+        use_python_runtime: bool = _defaults.USE_PYTHON_RUNTIME,
+        use_fast_partitioner: bool = _defaults.USE_FAST_PARTITIONER,
+        enable_experimental_decompositions: bool = _defaults.ENABLE_EXPERIMENTAL_DECOMPOSITIONS,
+        dryrun: bool = _defaults.DRYRUN,
+        hardware_compatible: bool = _defaults.HARDWARE_COMPATIBLE,
+        timing_cache_path: str = _defaults.TIMING_CACHE_PATH,
+        **kwargs: Any,
+    ) -> None:
+
+        self.pytorch_model = _make_refit_change_trigger(pytorch_model)
+        self.original_model = pytorch_model
+        # Process settings
+        self.gm: Any = None
+        self.exp_program: Any = None
+        self.inputs: Optional[tuple[Any, ...]] = None
+        device = to_torch_tensorrt_device(device)
+        enabled_precisions = {dtype._from(p) for p in enabled_precisions}
+        if not make_refitable:
+            logger.warning(
+                "'make_refitable' has to be True for a MutableTorchTensorRTModule."
+            )
+            make_refitable = True
+        compilation_options = {
+            "enabled_precisions": (
+                enabled_precisions
+                if enabled_precisions
+                else _defaults.ENABLED_PRECISIONS
+            ),
+            "debug": debug,
+            "device": device,
+            "assume_dynamic_shape_support": assume_dynamic_shape_support,
+            "workspace_size": workspace_size,
+            "min_block_size": min_block_size,
+            "torch_executed_ops": (
+                torch_executed_ops if torch_executed_ops is not None else set()
+            ),
+            "pass_through_build_failures": pass_through_build_failures,
+            "max_aux_streams": max_aux_streams,
+            "version_compatible": version_compatible,
+            "optimization_level": optimization_level,
+            "use_python_runtime": use_python_runtime,
+            "truncate_double": truncate_double,
+            "use_fast_partitioner": use_fast_partitioner,
+            "num_avg_timing_iters": num_avg_timing_iters,
+            "enable_experimental_decompositions": enable_experimental_decompositions,
+            "require_full_compilation": require_full_compilation,
+            "disable_tf32": disable_tf32,
+            "sparse_weights": sparse_weights,
+            "make_refitable": make_refitable,
+            "engine_capability": engine_capability,
+            "dla_sram_size": dla_sram_size,
+            "dla_local_dram_size": dla_local_dram_size,
+            "dla_global_dram_size": dla_global_dram_size,
+            "dryrun": dryrun,
+            "hardware_compatible": hardware_compatible,
+            "timing_cache_path": timing_cache_path,
+        }
+
+        self.settings = CompilationSettings(**compilation_options)
+
+    def load_state_dict(self, state_dict: dict[str, Any]) -> None:
+        RefitState.set_state(RefitFlag.NEEDS_REFIT)
```

Comment:
What happens if there is more than one mutable module?

Reply:
You are right!

Reply:
Fixed